### PR TITLE
Fix DB schema migrations

### PR DESCRIPTION
**Pull request by apiraino**
- Fix a wrong migration introduced in #1781
- Clarify how DB migrations should be added to the triagebot

r? @jackh726

### CODE REVIEW

File: src/db.rs [hunk 1]
```diff
@@ -332,7 +335,8 @@ CREATE table review_prefs (
     assigned_prs INT[] NOT NULL DEFAULT array[]::INT[]
 );",
     "
-CREATE EXTENSION intarray;
-CREATE UNIQUE INDEX review_prefs_user_id ON review_prefs(user_id);
+CREATE EXTENSION IF NOT EXISTS intarray;",
+    "
+CREATE UNIQUE INDEX IF NOT EXISTS review_prefs_user_id ON review_prefs(user_id);
```
Comment (apiraino) [inline src/db.rs:340]: I can't check the schema of the triagebot in prod but the clauses `IF NOT EXISTS` should ensure nothing bad happens when applied.

**Comment by jackh726** [comment]
I think we should get eyes on this from @Mark-Simulacrum as I'm not very familiar with how migrations work and what the current state of the db is.

**Comment by ehuss** [comment]
Will this fix the errors about `function sort(integer[]) does not exist`?

**Comment by apiraino** [comment]
> Will this fix the errors about `function sort(integer[]) does not exist`?

@ehuss Yes and now I know why. Without this patch the migration would fail:
```
$ cargo run --bin triagebot
...
Failed to run server: database migrations

Caused by:
    0: executing 12th migration
    1: db error: ERROR: cannot insert multiple commands into a prepared statement
    2: ERROR: cannot insert multiple commands into a prepared statement
```

unsure why it didn't happen (or wasn't catched) during the triagebot deployment at the time :man_shrugging:

**Review by jackh726 — APPROVED** [review]
After talking with @Kobzol seems like this is fine

**Pull request merged**